script.js

```diff
@@ -1,4 +1,184 @@
+  var deck = [];
+  var suitIndex = 0;
+  var suits = ["hearts", "diamonds", "clubs", "spades"];
+  while (suitIndex < suits.length) {
+    var currentSuit = suits[suitIndex];
+    //console.log("current suit: " + currentSuit);
+    var rankCounter = 1;
+    while (rankCounter <= 13) {
+      var cardName = rankCounter;
+      if (cardName == 1) {
+        cardName = "ace";
+      } else if (cardName == 11) {
+        cardName = "jack";
+      } else if (cardName == 12) {
+        cardName = "queen";
+      } else if (cardName == 13) {
+        cardName = "king";
+      }
+      var card = {
+        name: cardName,
+        suit: currentSuit,
+        rank: rankCounter,
+      };
+      //console.log("rank: " + rankCounter);
+      deck.push(card);
+      rankCounter += 1;
+    }
+    suitIndex += 1;
+  }
+  return deck;
+};
+
+// shuffle deck 
+var getRandomIndex = function (max) {
+  return Math.floor(Math.random() * max);
+};
+
+// shuffle the deck and save it in a new variable shuffledDeck
+// to communicate that we have shuffled the deck.
+var shuffleCards = function (cardDeck) {
+  var currentIndex = 0;
+  while (currentIndex < cardDeck.length) {
+    var randomIndex = getRandomIndex(cardDeck.length);
+    var randomCard = cardDeck[randomIndex];
+    var currentCard = cardDeck[currentIndex];
+    cardDeck[currentIndex] = randomCard;
+    cardDeck[randomIndex] = currentCard;
+    currentIndex = currentIndex + 1;
+  }
+  return cardDeck;
+};
+
+var deck = makeDeck();
+var shuffledDeck = shuffleCards(deck);
+
+
+//function to take a hand and come up with points
+//add rank
+// count aces
+// write the score
+var calculate = function(hand){
+  var currentScore = 0;
+  var cardIndex = 0;
+  var handLength = hand.length;
+  var gotAce = 0;
+  while (cardIndex < handLength){
+    var currentCard = hand[cardIndex]
+    var cardScore = currentCard.rank
+    if (cardScore == 1){
+      gotAce += 1;
+    }
+    currentScore += Math.min(10,cardScore)
+    cardIndex += 1
+  }
+  if (gotAce > 0 & currentScore <= 11){
+    currentScore += 10;
+  }
+  return currentScore
+}
+
+//banker to decide whether to draw or not
+var dealerActions = function(){
+  var currentScore = calculate(computerHand);
+  console.log(currentScore)
+  while (currentScore < 17){
+    var newCard = shuffledDeck.pop();
+    computerHand.push(newCard);
+    currentScore = calculate(computerHand);
+    console.log(currentScore)
+  }
+}
+
+
+// if user choose stand, compare card with computer
+// show results and end game.
+
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+  // Submit to deal 2 cards to computer and player
+  if (gameMode == 'deal2Cards'){
+    var playerCard1 = shuffledDeck.pop();
+    var playerCard2 = shuffledDeck.pop();
+    playerHand.push(playerCard1,playerCard2);
+    var computerCard1 = shuffledDeck.pop();
+    var computerCard2 = shuffledDeck.pop();
+    computerHand.push(computerCard1,computerCard2);
```

Comment:
You could create a helper function that helps add cards to the player and computer hand